File: src/java/main/commands/RichJson_env.java
package commands;

import core.*;
import other.RichJsonEnvironment;

import java.util.Map;

public class RichJson_env implements RichJsonCommand {
    @Override
    @SuppressWarnings("unchecked")
    public Object execute(RichJsonParser parser, RichJsonContext context) throws Exception {
        if (context.currentMember instanceof Map) {
            RichJsonEnvironment.addEnvironmentVariables((Map<String, Object>) context.currentMember);
            return context.currentMember;
        }

        String memberStr = (String) context.currentMember;
        String[] ref = memberStr.split(RichJsonConstants.COMMAND_PATH_DELIMITER, 2);
        String firstRef = ref[0];

        if (RichJsonEnvironment.env.containsKey(firstRef)) {
            Object envVal = RichJsonEnvironment.env.get(firstRef);
            context.currentMember = envVal;

            // Setzt temporär den Root auf die Env-Variable für interne Auflösungen
            context.root = (envVal instanceof Map) ? envVal : Map.of();
            context.currentAddress = parser.cache.resolveAddress(context.root);

            // Rekursives Parsen des Inhalts der Env-Variable
            context.currentMember = parser.__parseRichJsonInMember();
            RichJsonEnvironment.env.put(firstRef, context.currentMember);

            if (ref.length == 2) {
                context.currentMember = ref[1];
                return RichJsonCommandHolder.executeCommand("ref", parser, context);
            }
            return context.currentMember;
        } else {
            throw new RuntimeException("Environment variable '" + memberStr + "' does not exist.");
        }
    }
}

File: src/c#/main/commands/RichJson_env.cs
using System;
using System.Collections.Generic;
using System.Linq;
using main.core;
using main.other;

namespace main.commands
{
    public class RichJson_env
    {
        public object Execute(RichJsonParser parser, RichJsonContext context)
        {
            if (context.CurrentMember is IDictionary<string, object> memberMap)
            {
                RichJsonEnvironment.AddEnvironmentVariables(memberMap as Dictionary<string, object> ?? new Dictionary<string, object>(memberMap));
                return context.CurrentMember;
            }

            string memberStr = (string)context.CurrentMember;
            string[] @ref = memberStr.Split(new[] { RichJsonConstants.COMMAND_PATH_DELIMITER }, 2, StringSplitOptions.None);
            string firstRef = @ref[0];

            if (RichJsonEnvironment.Env.ContainsKey(firstRef))
            {
                object envVal = RichJsonEnvironment.Env[firstRef];
                context.CurrentMember = envVal;

                var prevRoot = context.Root;
                context.Root = (envVal is IDictionary<string, object>) ? envVal : new Dictionary<string, object>();
                context.CurrentAddress = parser.Cache.ResolveAddress(context.Root);

                context.CurrentMember = parser.ParseRichJsonInMember();
                RichJsonEnvironment.Env[firstRef] = context.CurrentMember;

                if (@ref.Length == 2)
                {
                    context.CurrentMember = @ref[1];
                    return RichJsonCommandHolder.ExecuteCommand("ref", parser, context);
                }
                context.Root = prevRoot;
                return context.CurrentMember;
            }
            else
            {
                throw new Exception("Environment variable or path '" + memberStr + "' does not exist.");
            }
        }
    }
}